compaction/compaction_manager.cc

```diff
@@ -293,6 +294,8 @@ class compaction_manager::major_compaction_task : public compaction_manager::tas
     // it cannot be the other way around, or minor compaction for this table would be
     // prevented while an ongoing major compaction doesn't release the semaphore.
     virtual future<> do_run() override {
+        co_await coroutine::switch_to(_cm._compaction_controller.sg());
+
```

Comment:
What about switching back?

Maybe all switching should be done by common code.

Reply:
The original scheduling group is restored when the coroutine completes.
See scylladb/seastar@6d061b7adaa482e38744c6a78790d71e482d82be